app/controllers/UploadController.scala

```diff
@@ -132,8 +132,11 @@ class UploadController @Inject()(authAction: AuthAction,
 
   def showuploadODSFile(scheme: String)
                        (implicit request: RequestWithOptionalEmpRef[AnyContent], hc: HeaderCarrier, messages: Messages): Future[Result] = {
-    clearErrorCache() map {
-      case false => getGlobalErrorPage(request, messages)
+
+    for (
+      clearedSuccessfully <- clearErrorCache()
+    ) yield {
+      if (!clearedSuccessfully) getGlobalErrorPage(request, messages)
```

Comment:
this looks fine but there's coverage missing for `getGlobalErrorPage(request, messages)`. It might be worth adding a test to ensure this code works as intended

Reply:
Good call, this was not working as intended (and it wasn't before these changes either) - have updated and added test in my latest commit